test/test_torch.py

```diff
@@ -11148,31 +11131,26 @@ def test_exponential(self, device, dtype):
                     torch.testing.get_all_complex_dtypes()))
     def test_exp(self, device, dtype):
         for v in (2, -2) + ((1j, 1 + 1j) if dtype.is_complex else ()):
-            if dtype == torch.bfloat16:
-                # Currently multiply a bfloat16 type with floating-point causes error:
-                #   RuntimeError: dtype != ScalarType::Undefined INTERNAL ASSERT FAILED at
-                #   "/pytorch/aten/src/ATen/native/TensorIterator.cpp":125, please report a bug to PyTorch.
-                # We skip bfloat16 for now, but we should fix it. https://github.com/pytorch/pytorch/issues/40580
-                if self.device_type == 'cpu' or self.device_type == 'cuda':
-                    with self.assertRaises(RuntimeError):
-                        torch.tensor(v, dtype=dtype, device=device) * torch.arange(18, device=device)
-                    return
-                elif self.device_type == 'xla':
-                    # Error:
-                    # Traceback (most recent call last):
-                    # File "/opt/conda/lib/python3.6/site-packages/torch/testing/_internal/common_device_type.py",
-                    # line 241, in instantiated_test
-                    #   result = test(self, device_arg, dtype)
-                    # File "/var/lib/jenkins/workspace/xla/test/../../test/test_torch.py", line 11062, in test_exp
-                    #   self.compare_with_numpy(torch.exp, np.exp, a)
-                    # File "/opt/conda/lib/python3.6/site-packages/torch/testing/_internal/common_utils.py", line 878,
-                    # in compare_with_numpy
-                    #   a = tensor_like.detach().cpu().numpy()
-                    # TypeError: Got unsupported ScalarType BFloat16
-                    return
+            if dtype == torch.bfloat16 and self.device_type == 'xla':
```

Comment:
I would cut the error message here (I didn't love that it was added in the first place). Instead of returning, though, would you assert that this fails?

Reply:
It turns out that this has already been asserted a few lines below this block. I've removed the whole block now.